### ISSUE TITLE
unreachable_code_linter() triggers on code after a switch with an error

### ISSUE THREAD

**Issue by MichaelChirico** [jimhester/lintr#3084]
```r
lint(linters = unreachable_code_linter(), text = '
foo <- function(x) {
  y = switch(x, a = 1, b = 2, stop("bad x"))
  return(x + y)
}
')
# <text>:3:44: warning: [unreachable_code_linter] Remove code and comments coming after stop().
#   y = switch(x, a = 1, b = 2, stop("bad x"))
#                                            ^
```

**Comment by MichaelChirico** [comment]
Ditto for named `switch()` entries:

```r
lint(linters = unreachable_code_linter(), text = '
foo <- function(x) {
  y = switch(x, a = 1, b = stop("no b"), c = 2)
  return(x + y)
}
')
# <text>:3:40: warning: [unreachable_code_linter] Remove code and comments coming after stop().
#   y = switch(x, a = 1, b = stop("no b"), c = 2)
#                                        ^
```

**Comment by MichaelChirico** [comment]
This also applies to `stop()` in a compound return statement:

```r
lint(linters = unreachable_code_linter(), text = '
foo <- function() {
  return(stop("bad"))
}
')
# <text>:3:21: warning: [unreachable_code_linter] Remove code and comments coming after stop().
#   return(stop("bad"))
#                     ^
```

NB: `{` is required

**Comment by MichaelChirico** [comment]
And to `tryCatch()`:

```r
lint(linters = unreachable_code_linter(), text = R"(
foo <- function() {
  tryCatch(
     return(.foo()),
    error = function(cond) {}
  )
}
)")
# <text>:4:20: warning: [unreachable_code_linter] Remove code and comments coming after return().
#      return(.foo()),
#                    ^
```
